front: stop showing arrival warning due to rounding errors.

anisometropie opened the pull request:
the simulation result output table oftens was displaying scheduled arrival time and calculated arrival time with a difference of one second, due to rounding errors during interpolation.

Values such that | scheduleArrival - calculatedArrival | <= 1 second are now displayed both as scheduleArrival, and no warning is shown

fixes #8185

woshilapin commented on front/src/modules/timesStops/hooks/useOutputTableData.ts:
```diff
@@ -89,11 +90,15 @@ function useOutputTableData(
             selectedTrainSchedule,
             pathStepId
           );
+          const calculatedArrival = secToHoursString(opPoint.time, true);
```
Why converting to a string, and then comparing typed datetime (don't know exactly what type it is but I guess `dayjs` return something typed)? Wouldn't it be more logical to compare first datetimes directly?

anisometropie replied:
I’m now comparing the numbers directly before converting the TimeString.
added some tests as well

woshilapin commented on front/src/modules/timesStops/hooks/useOutputTableData.ts:
```diff
@@ -89,11 +90,15 @@ function useOutputTableData(
             pathStepId
           );
+          const calculatedArrival = secToHoursString(opPoint.time, true);
+          const arrivalValuesAreClose =
+            Math.abs(compareTimeString(scheduleData.arrival, calculatedArrival)) <= 1000;
```
How about auto-documented code by naming this hardcoded number (without it, it looks like a non-explained arbitrary value). I'm open to any other naming, I just didn't came up with a good idea.

```suggestion
          const MAXIMAL_ROUNDING_ERROR_IN_MS = 1000;
          const arrivalValuesAreClose =
            Math.abs(compareTimeString(scheduleData.arrival, calculatedArrival)) <= 1000;
```

anisometropie replied:
added a named constant

woshilapin reviewed (changes requested):
Nice work tackling that bug.

At the very least, it would be good to add some unit tests so this fix never regress.

anisometropie committed:
front: stop showing arrival not honored warning due to rounding errors.

the simulation result output table oftens was displaying scheduled arrival time and calculated arrival time with a difference of one second, due to rounding errors during interpolation.

Values such that | scheduleArrival - calculatedArrival | <= 1 second are not displayed both as scheduleArrival, and no warning is shown

clarani reviewed (approved):
LGTM

Yohh commented on front/src/modules/timesStops/helpers/scheduleData.ts:
```diff
@@ -0,0 +1,35 @@
+  }
+  const startTimeSeconds = datetime2sec(new Date(startTime));
+  // relative value, number of seconds since startTime
+  const arrivalSeconds = schedule.arrival
+    ? startTimeSeconds + ISO8601Duration2sec(schedule.arrival)
+    : null;
+  const stopForSeconds = schedule.stop_for ? ISO8601Duration2sec(schedule.stop_for) : null;
+  const departure =
+    arrivalSeconds && stopForSeconds ? startTimeSeconds + arrivalSeconds + stopForSeconds : null;
+  return {
+    arrival: arrivalSeconds,
+    departure,
+    stopFor: stopForSeconds,
```
```suggestion
  const arrival = schedule.arrival
    ? startTimeSeconds + ISO8601Duration2sec(schedule.arrival)
    : null;
  const stopFor = schedule.stop_for ? ISO8601Duration2sec(schedule.stop_for) : null;
  const departure = arrival && stopFor ? startTimeSeconds + arrival + stopFor : null;
  return {
    arrival,
    departure,
    stopFor,
```

Yohh replied:
it was maybe to explicitly show that there is seconds, but the function name seems enough

clarani replied:
I would rather keep it like this, because otherwise we don't know at first sight what we are manipulating. Is it ok for you ?

woshilapin reviewed (approved):
Thank you for addressing the comments and adding some tests.
